src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -373,6 +421,7 @@ static bool equalMetadataValues(BSONObj& md1, BSONObj& md2)
 */
 static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2)
```

Comment:
Maybe... Although I'd suggest refactor in that way in the PR only if that is an easy task. Otherwise, let's mark with a FIXME and let it as is. @crbrox , as you prefer :)